jyotisha/panchaanga/temporal/festival/applier/solar.py

```diff
@@ -22,6 +25,23 @@ def assign_all(self):
     self.assign_agni_nakshatra()
 
 
+  def assign_sankranti_punyakaala(self):
+    PUNYA_KAALA = {1: (10, 10), 2: (16, 16), 3: (0, 60), 4: (30, 0), 5: (16, 16), 6: (0, 60),
+                   7: (10, 10), 8: (16, 16), 9: (0, 60), 10: (0, 20), 11: (16, 16), 12: (0, 60)}
+    SANKRANTI_PUNYAKALA_NAMES = {1: "meSa-viSu", 2: "viSNupadI", 3: "SaDazIti", 4: "kaTaka-saGkrAnti",
+      5: "viSNupadI", 6: "SaDazIti", 7: "tulA-viSu", 8: "viSNupadI",
+      9: "SaDazIti", 10: "makara-saGkrAnti", 11: "viSNupadI", 12: "SaDazIti"}
+
+    for d in range(self.panchaanga.duration_prior_padding, self.panchaanga.duration + 1):
+      if self.daily_panchaangas[d].solar_sidereal_date_sunset.month_transition is not None:
+        punya_kaala_str = SANKRANTI_PUNYAKALA_NAMES[self.daily_panchaangas[d + 1].solar_sidereal_date_sunset.month] + '-puNyakAlaH'
+        jd_transition = self.daily_panchaangas[d].solar_sidereal_date_sunset.month_transition
+        # TODO: convert carefully to relative nadikas!
+        punya_kaala_start_jd = jd_transition - PUNYA_KAALA[self.daily_panchaangas[d + 1].solar_sidereal_date_sunset.month][0] * 1/60
+        punya_kaala_end_jd = jd_transition + PUNYA_KAALA[self.daily_panchaangas[d + 1].solar_sidereal_date_sunset.month][1] * 1/60
+        self.daily_panchaangas[d].festival_id_to_instance[punya_kaala_str] = ( FestivalInstance(name=punya_kaala_str, interval=Interval(jd_start=punya_kaala_start_jd, jd_end=punya_kaala_end_jd)))
```

Comment:
Also, please utilize this new method (just checked in) : https://github.com/sanskrit-coders/jyotisha/commit/17fdf2a47b4c14cd17108fba932bb8cf307fc2a2 to add festival instances (handles both the relevant places which need to be updated).

Reply:
This seems to not display the time. Have included it as commented line.